File: backend/src/main/java/com/documed/backend/additionalservices/AdditionalService.java
package com.documed.backend.additionalservices;

import com.documed.backend.attachments.Attachment;
import com.documed.backend.notifications.Notification;
import com.documed.backend.services.Service;
import com.documed.backend.users.User;
import com.documed.backend.visits.VisitStatus;
import java.util.Date;
import java.util.List;
import lombok.Data;
import lombok.NonNull;

@Data
public class AdditionalService {
  private final int id;
  private String description;
  @NonNull private Date date;
  @NonNull private User fulfiller;
  @NonNull private Service service;
  @NonNull private VisitStatus status;
  private List<Notification> notifications;
  private List<Attachment> attachments;
}

File: frontend/src/shared/types/AdditionalService.ts
import { AttachmentLite } from './Attachment';
import { VisitStatus } from './enums';
import { NotificationLite } from './Notification';
import { ServiceLite } from './Service';
import { UserLite } from './User';

export interface AdditionalService extends AdditionalServiceLite {
  fulfiller: UserLite;
  service: ServiceLite;
  notifications: NotificationLite[] | null;
  attachments: AttachmentLite[] | null;
}

export interface AdditionalServiceLite {
  id: number;
  description: string | null;
  date: Date;
  status: VisitStatus;
}
